sql/sql_parse.cc

```diff
@@ -5153,9 +5153,36 @@ mysql_execute_command(THD *thd)
       my_ok(thd);
     break;
   case SQLCOM_BACKUP_LOCK:
-    if (check_global_access(thd, RELOAD_ACL))
-      goto error;
-    /* first table is set for lock. For unlock the list is empty */
+    if (check_global_access(thd, RELOAD_ACL, true))
+    {
+      /* In case there is no global privilege, check DB privilege for BACK LOCK.*/
+      if (first_table)
+      {
+        if (check_single_table_access(thd, LOCK_TABLES_ACL, first_table, true))
+        {
+#ifndef NO_EMBEDDED_ACCESS_CHECKS
+          char command[30];
+          get_privilege_desc(command, sizeof(command), RELOAD_ACL|LOCK_TABLES_ACL);
+          my_error(ER_SPECIFIC_ACCESS_DENIED_ERROR, MYF(0), command);
+#endif
+          goto error;
+        }
+      }
+      else // BACKUP UNLOCK
+      {
+        if (!thd->mdl_backup_lock)
+        {
+#ifndef NO_EMBEDDED_ACCESS_CHECKS
+          char command[30];
+          get_privilege_desc(command, sizeof(command), RELOAD_ACL);
```

Comment:
Would be both `RELOAD_ACL|LOCK_TABLES_ACL`. `command` may need to be bigger.

What error under `RELOAD_ACL` happens when you `BACKUP UNLOCK` without a table `BACKUP LOCK`ed? This should be the same.

Reply:
There will be no error. Basically when you call `backup lock` it will first call `unlock` and process further.
Do you want to add that specific test case too?

Reply:
rough thought while rather sleepy says yes, if your code implements it a case to show its behaviour is good.

Reply:
Unlocking without existing backup lock:

with ACL_RELOAD -  no error or warning

new behaviour - with LOCK_TABLES

Existing:
```
MariaDB [test]> backup unlock
    -> ;
ERROR 1227 (42000): Access denied; you need (at least one of) the RELOAD privilege(s) for this operation
```

New depends which DB.  `mdl_backup_lock`/`mdl_context`/`mdl_context.owner` won't give the DB.  `thd->db` as the current DB is messy.
 
So document behaviour in the commit message/comment that we know there's an unconsistency with the `backup unlock` in the case where no RELOAD privs, and no table is locked, because we can't easily determine if `lock tables` can apply.